app/components/sidebar.tsx

```diff
@@ -0,0 +1,167 @@
+"use client";
+
+require("../polyfill");
+
+import { useState, useEffect } from "react";
+
+import { IconButton } from "./button";
+import styles from "./home.module.scss";
+
+import SettingsIcon from "../icons/settings.svg";
+import GithubIcon from "../icons/github.svg";
+import CvIcon from "../icons/cv.svg";
+import RobotIcon from "../icons/robotgpt.svg";
+
+import BotIcon from "../icons/bot.svg";
+import AddIcon from "../icons/add.svg";
+import LoadingIcon from "../icons/three-dots.svg";
+import CloseIcon from "../icons/close.svg";
+
+import { useChatStore } from "../store";
+import Locale from "../locales";
+
+import { REPO_URL } from "../constant";
+
+export function Loading(props: { noLogo?: boolean }) {
+  return (
+    <div className={styles["loading-content"]}>
+      {!props.noLogo && <BotIcon />}
+      <LoadingIcon />
+    </div>
+  );
+}
+
+function useSwitchTheme() {
+  const config = useChatStore((state) => state.config);
+
+  useEffect(() => {
+    document.body.classList.remove("light");
+    document.body.classList.remove("dark");
+
+    if (config.theme === "dark") {
+      document.body.classList.add("dark");
+    } else if (config.theme === "light") {
+      document.body.classList.add("light");
+    }
+
+    const metaDescriptionDark = document.querySelector(
+      'meta[name="theme-color"][media]',
+    );
+    const metaDescriptionLight = document.querySelector(
+      'meta[name="theme-color"]:not([media])',
+    );
+
+    if (config.theme === "auto") {
+      metaDescriptionDark?.setAttribute("content", "#151515");
+      metaDescriptionLight?.setAttribute("content", "#fafafa");
+    } else {
+      const themeColor = getComputedStyle(document.body)
+        .getPropertyValue("--theme-color")
+        .trim();
+      metaDescriptionDark?.setAttribute("content", themeColor);
+      metaDescriptionLight?.setAttribute("content", themeColor);
+    }
+  }, [config.theme]);
+}
+
+const useHasHydrated = () => {
+  const [hasHydrated, setHasHydrated] = useState<boolean>(false);
+
+  useEffect(() => {
+    setHasHydrated(true);
+  }, []);
+
+  return hasHydrated;
+};
+
+export default function Sidebar(props: {
+  title: string;
+  subTitle: string;
+  setOpenSettings?: (newState: boolean) => void;
+  children: React.ReactNode;
+}) {
+  const [createNewSession, currentIndex, removeSession] = useChatStore(
+    (state) => [
+      state.newSession,
+      state.currentSessionIndex,
+      state.removeSession,
+    ],
+  );
+  const chatStore = useChatStore();
```

Comment:
Det er alltid litt trade-offs her, men det kan jo være hensiktsmessig å la dette sendes inn fra toppen og la Sidebar ikke ha ansvar for dette. Kan spørre seg "om en kaller noe Sidebar, er det naturlig at den har ansvaret for å opprette ny session"? I dette tilfellet heller jeg mot at det er litt unaturlig.